### PR TITLE
[explorer] Pagination Redesign and Refactoring

### PR DESCRIPTION
**Pull request by Andrew47**
* Redesigns the Pagination following: https://www.figma.com/file/OzLaRFzevjxdQAbybWEZk0/Sui-Visualize?node-id=2979%3A59684.
* Refactors Pagination down to one component as discussed here: https://github.com/MystenLabs/sui/pull/3157

# Video showing redesign in action:

https://user-images.githubusercontent.com/11377188/180053244-0672f6f1-1e63-4df7-930e-3dd407458e36.mp4

This video goes through the following order:

1. Demo of pagination looks on Desktop
2. Demo on Mobile (Galaxy Fold emulation)

### CODE REVIEW

File: explorer/client/src/components/pagination/Pagination.tsx
Comment (Jibz1) [inline explorer/client/src/components/pagination/Pagination.tsx:6]: Why add the tab footer to the pagination module

**Review by Jibz1 — CHANGES_REQUESTED** [review]

**Comment by Jibz1** [comment]
This was done already 
Screen Shot 2022-07-20 at 2 45 38 PM

**Comment by mystie711** [comment]
👍🏽 ui, interaction looking good.

File: explorer/client/src/components/pagination/Pagination.tsx
Comment (Andrew47) [reply explorer/client/src/components/pagination/Pagination.tsx:6]: This fixed an issue with alligning the text across the footer.

Comment (Jibz1) [reply explorer/client/src/components/pagination/Pagination.tsx:6]: They are separate items, there are multiple areas where the tap footer is not path of the pagination

**Comment by Andrew47** [comment]
@Jibz1 --> When reviewing this PR, please bear in mind that the pagination UX/UI follows the Figma specifications detailed here: https://www.figma.com/file/OzLaRFzevjxdQAbybWEZk0/Sui-Visualize?node-id=2979%3A59684. These are not currently implemented in the pagination in main.

**Comment by Andrew47** [comment]
@Jibz1 --> Thank you for highlighting that pagination has been applied to the display of Modules in the Transaction Results Page. Please could you provide the transaction ID associated with these screenshots.

**Comment by Jibz1** [comment]
> @Jibz1 --> Thank you for highlighting that pagination has been applied to the display of Modules in the Transaction Results Page. Please could you provide the transaction ID associated with these screenshots.

https://explorer.devnet.sui.io/transactions/zpVKn9Ddxc8wMH3TEaU0S0vS9w386Z3GVMrvApR%2BfhA%3D

File: explorer/client/src/components/pagination/Pagination.tsx
Comment (Jibz1) [reply explorer/client/src/components/pagination/Pagination.tsx:6]: Screen Shot 2022-07-20 at 3 14 53 PM
Screen Shot 2022-07-20 at 3 15 07 PM

**Comment by Andrew47** [comment]
@Jibz1 --> I've fixed the pagination for the display of modules in the Transaction Results Page.

File: explorer/client/src/components/pagination/Pagination.tsx
Comment (Andrew47) [reply explorer/client/src/components/pagination/Pagination.tsx]: I've removed `TabFooter` from Pagination.

**Comment by Andrew47** [comment]
@Jibz1 / @666lcz --> the feedback has been incorporated; this is ready for a second review.

**Comment by Jibz1** [comment]
> @Jibz1 / @666lcz --> the feedback has been incorporated; this is ready for a second review.

thanks, @Andrew47. Is the issue highlighted in the video been fixed? where once you go to the last page and change the number of items per page to a larger one, the pagination arrows stop working

**Comment by Andrew47** [comment]
> > @Jibz1 / @666lcz --> the feedback has been incorporated; this is ready for a second review.
> 
> thanks, @Andrew47. Has the issue highlighted in the video been fixed? The pagination arrows stop working when you go to the last page and change the number of items per page to a larger one.

Yes this is fixed

File: explorer/client/src/components/transaction-card/RecentTxCard.tsx
Comment (Jibz1) [inline explorer/client/src/components/transaction-card/RecentTxCard.tsx:36]: Just one small thing, can we keep this at 15? The Figma design on the home page is 15 transactions

**Review by Jibz1 — APPROVED** [review]

File: explorer/client/src/components/transaction-card/RecentTxCard.tsx
Comment (Andrew47) [reply explorer/client/src/components/transaction-card/RecentTxCard.tsx:36]: Looking at the deployed website, although the constant says '15', '20' items are displayed because the filter is overriding this constant. If we did enforce 15 per page this would potentially be confusing in that the filter says "20 per page". In the next update to the Home Page design, feel free to change this to 15 but make sure that this constant is not overridden.

**Pull request merged**
